libyul/AST.h

```diff
@@ -35,54 +35,63 @@ namespace solidity::yul
 
 using Type = YulString;
 
-struct TypedName { langutil::SourceLocation location; YulString name; Type type; };
+struct YulAstDebugData {
+	explicit YulAstDebugData(langutil::SourceLocation _yulLocation = {}, langutil::SourceLocation _sollLocation = {}):
+		yulLocation(std::move(_yulLocation)),
+		solLocation(std::move(_sollLocation))
+	{}
+	langutil::SourceLocation yulLocation;
+	langutil::SourceLocation solLocation;
```

Comment:
irLocation and originLocation? sourceLocation? originalLocation?

Reply:
went with `irLocation` and `originLocation`